typescript/src/generator.ts

```diff
@@ -66,7 +68,7 @@ export class Generator {
   }
 
   private async readGrpcServiceConfig(parameter: string) {
-    const match = parameter.match(/^["']?grpc-service-config=([^"]+)["']?$/);
+    const match = parameter.match(/^["']?grpc-service-config=([^"]+)["']?/);
```

Comment:
if both `grpc-service-config` and `package-name` are passed, this won't work (the regex will take the whole string up to and including the `package-name` parameter).

I think at this point, since we have two parameters (and may have more), we need to parse it in some more accurate way. Let's make an assumption that parameter values cannot have spaces, then we can have a function that parses this string and returns a map:

```ts
function parseParameterString(parameters: string) {
  const params = parameters.split(' ');
  ...
  // return an object of key-value pairs
}
```

and then just parse it once and take `params["grpc-service-config"]`, `params["package-name"]`, etc.  What do you think?

Reply:
yes, I noticed the same thing that >(the **former** regex will take the whole string up to and including the package-name parameter). 
so I modified a little bit the regex, now match[1] only takes the correct argument `example.json` and `@google-cloud/text-to-speech`.  Test case `extra-option.ts` takes both `grpc-service-config` and `package-name` which works fine. 

But I think making key-value pair makes perfect sense for adding more options to generator.

Reply:
`^` at the beginning forces the grpc service config parameter to go first. maybe remove it?

Reply:
I will make a map for the parameters which makes easier to add more arguments, and more readable. thanks for the suggestion!